### ISSUE TITLE
InputObject Invoke-DscConfigurationByRoles

### ISSUE THREAD

**Issue by MisterTK** [Duffney/InvokeDSC#8]
This is great, exactly what I've been looking for. 

My goal is to avoid files on the local systems all together. I forked this just to add InputObject param to Get-ModuleFromConfiguration and Invoke-DscConfiguration, it's working well. Simple thing to add, but I can open pull request if you want. 

I haven't thought through how Invoke-DscConfigurationByRoles could be modified to support recursing through several objects in a url and processing them.

**Comment by duffney** [comment]
absolutely put in a PR to add the inputobject, I didn't do that out of laziness and should have to support your use case of no files on disk. It's awesome someone else found this useful! I've made some breaking changes to the json format to declare objects instead of arrays. This allowed you to use Octopus Deploy to transform variables and that's the context as to why I made the change. So before the PR make sure you've forked the latest build. As for the recursive roles, I'm not sure what the usecase would be. The cmdlet assumes you know the roles you want to invoke. Might be a good helper function depending on the scenario you've got.
